esmvaltool/diag_scripts/shared/_base.py

```diff
@@ -51,14 +53,85 @@ def get_diagnostic_filename(basename, cfg, extension='nc'):
     -------
     str:
         A valid path for saving a diagnostic data file.
-
     """
     return os.path.join(
         cfg['work_dir'],
         f"{basename}.{extension}",
     )
 
 
+def save_figure(basename, provenance, cfg, figure=None, close=True, **kwargs):
```

Comment:
We might consider the option `provenance=None` if the provenance information is already added to a `.nc` file and the plot is attached to it with `plot_file`, see comment below.

Reply:
@nielsdrost and I would like to remove the `plot_file` way of attaching provenance to a figure via a netcdf file in the future because it was confusing and didn't work well when the relation was not one plot - one netcdf file. It's much simpler to just write provenance for every output file. Therefore I would prefer not to add new functionality supporting this feature.